lms/djangoapps/verify_student/ssencrypt.py

```diff
@@ -51,68 +54,75 @@ def aes_encrypt(data, key):
 
 
 def aes_decrypt(encrypted_data, key):
     """
     Decrypt `encrypted_data` using `key`
     """
     cipher = aes_cipher_from_key(key)
-    padded_data = cipher.decrypt(encrypted_data)
+    decryptor = cipher.decryptor()
+    padded_data = decryptor.update(encrypted_data) + decryptor.finalize()
     return unpad(padded_data)
 
 
 def aes_cipher_from_key(key):
     """
-    Given an AES key, return a Cipher object that has `encrypt()` and
-    `decrypt()` methods. It will create the cipher to use CBC mode, and create
+    Given an AES key, return a Cipher object that has `encryptor()` and
+    `decryptor()` methods. It will create the cipher to use CBC mode, and create
     the initialization vector as Software Secure expects it.
     """
-    return AES.new(key, AES.MODE_CBC, generate_aes_iv(key))
+    return Cipher(AES(key), CBC(generate_aes_iv(key)), backend=default_backend())
 
 
 def generate_aes_iv(key):
     """
     Return the initialization vector Software Secure expects for a given AES
     key (they hash it a couple of times and take a substring).
     """
-    return md5(key + md5(key).hexdigest()).hexdigest()[:AES.block_size]
+    return md5(key + md5(key).hexdigest()).hexdigest()[:AES_BLOCK_SIZE_BYTES]
 
 
 def random_aes_key():
-    return Random.new().read(32)
+    return os.urandom(32)
 
 
 def pad(data):
     """ Pad the given `data` such that it fits into the proper AES block size """
-    bytes_to_pad = AES.block_size - len(data) % AES.block_size
-    return data + (bytes_to_pad * chr(bytes_to_pad))
+    padder = PKCS7(AES.block_size).padder()
+    return padder.update(data) + padder.finalize()
 
 
 def unpad(padded_data):
     """  remove all padding from `padded_data` """
-    num_padded_bytes = ord(padded_data[-1])
-    return padded_data[:-num_padded_bytes]
+    unpadder = PKCS7(AES.block_size).unpadder()
+    return unpadder.update(padded_data) + unpadder.finalize()
 
 
 def rsa_encrypt(data, rsa_pub_key_str):
     """
-    `rsa_pub_key` is a string with the public key
+    `rsa_pub_key` is a byte sequence with the public key
     """
-    key = RSA.importKey(rsa_pub_key_str)
-    cipher = PKCS1_OAEP.new(key)
-    encrypted_data = cipher.encrypt(data)
-    return encrypted_data
+    if rsa_pub_key_str.startswith(b'-----'):
+        key = serialization.load_pem_public_key(rsa_pub_key_str, backend=default_backend())
+    elif rsa_pub_key_str.startswith(b'ssh-rsa '):
+        key = serialization.load_ssh_public_key(rsa_pub_key_str, backend=default_backend())
+    else:
+        key = serialization.load_der_public_key(rsa_pub_key_str, backend=default_backend())
+    return key.encrypt(data, OAEP(MGF1(SHA1()), SHA1(), label=None))
 
 
 def rsa_decrypt(data, rsa_priv_key_str):
     """
     When given some `data` and an RSA private key, decrypt the data
     """
-    key = RSA.importKey(rsa_priv_key_str)
-    cipher = PKCS1_OAEP.new(key)
-    return cipher.decrypt(data)
+    if rsa_priv_key_str.startswith(b'-----'):
```

Comment:
Why is the ssh case missing here?

Reply:
Ah - think I see the answer above...

Reply:
I don't see the answer, what is it?

Reply:
The cryptography library doesn't provide support for loading ssh-style private keys (and we don't need that functionality anyway).